src/main/java/de/tum/in/www1/artemis/service/WebsocketMessagingService.java

```diff
@@ -78,18 +79,18 @@ public CompletableFuture<Void> sendMessage(String topic, Message<?> message) {
      *
      * @param topic   the destination to which subscription the message should be sent
      * @param message any object that should be sent to the destination (topic), this will typically get transformed into json
-     * @return a future that can be used to check if the message was sent successfully
+     * @return a future that can be used to check if the message was sent successfully or null if an error occurs
      */
+    @Nullable
     public CompletableFuture<Void> sendMessage(String topic, Object message) {
-        return CompletableFuture.runAsync(() -> {
-            try {
-                messagingTemplate.convertAndSend(topic, message);
-            }
-            catch (MessagingException ex) {
-                log.error("Error when sending message {} to topic {}", message, topic, ex);
-                throw ex;
-            }
-        }, asyncExecutor);
+        try {
+            return CompletableFuture.runAsync(() -> messagingTemplate.convertAndSend(topic, message), asyncExecutor);
+        }
+        // Note: explicitly catch ALL kinds of exceptions here and do NOT rethrow, because the actual task should NEVER be interrupted when the server cannot send WS messages
+        catch (Exception ex) {
+            log.error("Error when sending message {} to topic {}", message, topic, ex);
+        }
+        return null;
```

Comment:
This way the caller could work with the exception if needed.
```suggestion
        catch (Exception ex) {
            log.error("Error when sending message {} to topic {}", message, topic, ex);
            return CompletableFuture.failedFuture(ex);
        }
```

Reply:
good idea